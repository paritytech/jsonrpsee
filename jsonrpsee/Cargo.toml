[package]
name = "jsonrpsee"
description = "JSON-RPC crate"
version = "0.2.0"
authors = ["Parity Technologies <admin@parity.io>", "Pierre Krieger <pierre.krieger1708@gmail.com>"]
license = "MIT"
edition = "2018"
repository = "https://github.com/paritytech/jsonrpsee"
homepage = "https://github.com/paritytech/jsonrpsee"
documentation = "https://docs.rs/jsonrpsee"

[dependencies]
http-client = { path = "../http-client", version = "0.2.0", package = "jsonrpsee-http-client", optional = true }
http-server = { path = "../http-server", version = "0.2.0", package = "jsonrpsee-http-server", optional = true }
ws-client = { path = "../ws-client", version = "0.2.0", package = "jsonrpsee-ws-client", optional = true }
ws-server = { path = "../ws-server", version = "0.2.0", package = "jsonrpsee-ws-server", optional = true }
proc-macros = { path = "../proc-macros", version = "0.2.0", package = "jsonrpsee-proc-macros", optional = true }
utils = { path = "../utils", version = "0.2.0", package = "jsonrpsee-utils", optional = true }
types = { path = "../types", version = "0.2.0", package = "jsonrpsee-types", optional = true }

[features]
client = ["http-client", "ws-client"]
server = ["http-server", "ws-server", "utils"]
macros = ["proc-macros", "types"]
full = ["client", "server", "macros", "utils"]
