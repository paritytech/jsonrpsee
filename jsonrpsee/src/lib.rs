//! jsonrpsee wrapper crate.

/// JSON RPC HTTP client.
#[cfg(feature = "client")]
pub use http_client;

/// JSON RPC WebSocket client.
#[cfg(feature = "client")]
pub use ws_client;

/// JSON RPC HTTP server.
#[cfg(feature = "server")]
pub use http_server;

/// JSON RPC WebSocket server.
#[cfg(feature = "server")]
pub use ws_server;

/// Set of RPC methods that can be mounted to the server.
#[cfg(feature = "server")]
pub use utils::server::rpc_module::RpcModule;

/// Procedural macros for JSON RPC implementations.
#[cfg(feature = "macros")]
pub use proc_macros;

/// Common types used to implement JSON RPC server and client.
#[cfg(feature = "macros")]
pub use types;
