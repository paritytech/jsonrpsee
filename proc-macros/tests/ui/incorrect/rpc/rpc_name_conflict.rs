use jsonrpsee::{proc_macros::rpc, types::RpcResult};

// Associated items are forbidden.
#[rpc(client, server)]
pub trait MethodNameConflict {
	#[method(name = "foo")]
	async fn foo(&self) -> RpcResult<u8>;

	#[method(name = "foo")]
	async fn bar(&self) -> RpcResult<u8>;
}

fn main() {}
