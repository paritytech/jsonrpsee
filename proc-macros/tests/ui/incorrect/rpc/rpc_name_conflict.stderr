error: "foo" is already defined
  --> $DIR/rpc_name_conflict.rs:10:11
   |
10 |     async fn bar(&self) -> RpcResult<u8>;
   |              ^^^
